src/Workspaces/Remote/ServiceHub/Services/Host/RemoteHostService.cs

```diff
@@ -64,10 +64,17 @@ public RemoteHostService(Stream stream, IServiceProvider serviceProvider)
 
             if (TestData == null || !TestData.IsInProc)
             {
-                // Set this process's priority BelowNormal.
-                // this should let us to freely try to use all resources possible without worrying about affecting
-                // host's work such as responsiveness or build.
-                Process.GetCurrentProcess().PriorityClass = ProcessPriorityClass.BelowNormal;
+                try
+                {
+                    // Set this process's priority BelowNormal.
+                    // this should let us to freely try to use all resources possible without worrying about affecting
+                    // host's work such as responsiveness or build.
+                    Process.GetCurrentProcess().PriorityClass = ProcessPriorityClass.BelowNormal;
+                }
+                catch (Exception e) when (e is PlatformNotSupportedException || e is Win32Exception)
+                {
+                    // the runtime does not support changing process priority, so just ignore this
+                }
```

Comment:
i'd love if this was extracted into a TrySetPriorityClass that you used from both locations.  you can use the return value to determine if you should return nop-booster or not.

Reply:
Doable, was thinking about that, this part just popped up later in the testing phase for the remote workspace support.